### PR TITLE
Switching log lines in Core to new framework

### PR DESCRIPTION
**Pull request by bhariharan**

### CODE REVIEW

**Comment by huminzhi** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          2 Errors
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Tests have failed, see below for more information.</td>
    </tr>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Code coverage under minimum of 80%</td>
    </tr>
  </tbody>
</table>

### Tests: 

Classname | Name|
--- | ---|
CSFSalesforceOAuthRefreshTests | testRevokedToken|
CSFSalesforceOAuthRefreshTests | testRevokedToken|


## Current coverage for SalesforceSDKCore.framework is `48%`
Files changed | - | - 
--- | --- | ---
SFPasscodeViewController.m | `0%` | :skull:
SFMethodInterceptor.m | `0%` | :skull:
NSDictionary+SFAdditions.m | `0%` | :skull:
SFAbstractPasscodeViewController.m | `0%` | :skull:
SFInstrumentation.m | `0%` | :skull:
SFSDKAILTNPublisher.m | `0%` | :skull:
SFCrypto.m | `11%` | :skull:
UIDevice+SFHardware.m | `13%` | :skull:
NSData+SFAdditions.m | `19%` | :skull:
CSFSalesforceOAuthRefresh.m | `22%` | :skull:
SFSDKWebViewStateManager.m | `23%` | :skull:
SFJsonUtils.m | `23%` | :skull:
SFUserActivityMonitor.m | `26%` | :no_entry_sign:
SFPushNotificationManager.m | `28%` | :no_entry_sign:
SFSDKSalesforceAnalyticsManager.m | `31%` | :no_entry_sign:
UIColor+SFColors.m | `33%` | :no_entry_sign:
SFOAuthCoordinator.m | `36%` | :no_entry_sign:
SFInactivityTimerCenter.m | `38%` | :no_entry_sign:
SFSecurityLockout.m | `40%` | :no_entry_sign:
SFAuthenticationManager.m | `42%` | :no_entry_sign:
SFRootViewManager.m | `43%` | :no_entry_sign:
SFSHA256PasscodeProvider.m | `44%` | :no_entry_sign:
SFKeychainItemWrapper.m | `47%` | :no_entry_sign:
SFAuthErrorHandlerList.m | `50%` | :warning:
SFSDKTestRequestListener.m | `51%` | :warning:
CSFAction.m | `52%` | :warning:
SFUserAccount.m | `52%` | :warning:
SFDefaultUserAccountPersister.m | `55%` | :warning:
CSFInternalDefines.m | `55%` | :warning:
CSFInput.m | `56%` | :warning:
SalesforceSDKManager.m | `58%` | :warning:
CSFParameterStorage.m | `60%` | :warning:
SFKeyStore.m | `61%` | :warning:
SFDirectoryManager.m | `63%` | :warning:
SFPreferences.m | `63%` | :warning:
SFPasscodeManager.m | `64%` | :warning:
SFIdentityCoordinator.m | `68%` | :warning:
SFDecryptStream.m | `71%` | :warning:
CSFInputStreamElement.m | `72%` | :warning:
SFGeneratedKeyStore.m | `74%` | :warning:
SFPBKDF2PasscodeProvider.m | `75%` | :warning:
SFOAuthSessionRefresher.m | `78%` | :warning:
SFSDKCryptoUtils.m | `80%` | :warning:
SFPasscodeKeyStore.m | `81%` | :white_check_mark:
SFRestAPI.m | `82%` | :white_check_mark:
SFUserAccountManager.m | `82%` | :white_check_mark:
SFKeyStoreManager.m | `83%` | :white_check_mark:
SFOAuthCredentials.m | `83%` | :white_check_mark:
SFPasscodeProviderManager.m | `83%` | :white_check_mark:
SFOAuthKeychainCredentials.m | `86%` | :white_check_mark:
SFOAuthCrypto.m | `87%` | :white_check_mark:
SFSDKAsyncProcessListener.m | `88%` | :white_check_mark:
CSFAuthRefresh.m | `91%` | :white_check_mark:
SFCryptChunks.m | `92%` | :white_check_mark:

---
## Current coverage for SalesforceAnalytics.framework is `43%`
✅ *No files affecting coverage found*

---

> Powered by [xcov](https://github.com/nakiostudio/xcov)

<p align="right" data-meta="generated_by_CI-result">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFOAuthCoordinator.m [hunk 1]
```diff
@@ -185,18 +185,14 @@ - (void)authenticate {
     NSAssert(nil != self.delegate, @"cannot authenticate with nil delegate");
 
     if (self.authenticating) {
-        [self log:SFLogLevelDebug format:@"%@ Error: authenticate called while already authenticating. Call stopAuthenticating first.", NSStringFromSelector(_cmd)];
+        [SFSDKCoreLogger d:[self class] format:@"%@ Error: authenticate called while already authenticating. Call stopAuthenticating first.", NSStringFromSelector(_cmd)];
         return;
     }
-    if (self.credentials.logLevel < kSFOAuthLogLevelWarning) {
-        [self log:SFLogLevelDebug format:@"%@ authenticating as %@ %@ refresh token on '%@://%@' ...",
+    [SFSDKCoreLogger d:[self class] format:@"%@ authenticating as %@ %@ refresh token on '%@://%@' ...",
```
Comment (khawkins) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFOAuthCoordinator.m]: Is the plan to remove support for the OAuth-specific log level filtering in this timeframe?

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFOAuthCredentials.h [hunk 2]
```diff
@@ -25,17 +25,6 @@
 #import <Foundation/Foundation.h>
 
 /**
- @enum Logging levels to control the verbosity of log output based on the severity of the event being logged.
- */
-typedef NS_ENUM(NSUInteger, SFOAuthLogLevel) {
```
Comment (khawkins) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFOAuthCredentials.h]: Looks like 'yes' to my last question.  If so, we should honor the original log levels in our new logs.  Currently, we've left them all at 'Debug'.

Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFOAuthCredentials.h]: The default is `Debug` in debug builds and `Error` in prod builds. Are you suggesting `OAuth` should continue to have its own component (this breaks the design we seek to achieve with the new framework within the SDK)?

Comment (khawkins) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFOAuthCredentials.h]: Not that it should necessarily continue to have its own, but that we should translate its previous log levels into the new paradigm.  The old way we did that was to honor the OAuth-specific check in the form of an inline conditional, then log at the lowest level if that condition was met.  Here, we took out the conditional, but didn't consider its purpose, leaving 'Debug' logs for all of those.

So e.g. for the following code:
```obj-c
if (self.credentials.logLevel < kSFOAuthLogLevelWarning) {
    [self log:SFLogLevelDebug format:@"%@ host=%@ : path=%@", NSStringFromSelector(_cmd), url.host, url.path];
}
```
we should change to:
```obj-c
[SFSDKCoreLogger w:[self class] format:@"%@ host=%@ : path=%@", NSStringFromSelector(_cmd), url.host, url.path];
```

Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFOAuthCredentials.h]: Oh I see your point. I can update those levels.

Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFOAuthCredentials.h]: Done [here](https://github.com/forcedotcom/SalesforceMobileSDK-iOS/pull/2103/commits/faeadf27116f1e4e467ff36f48c01b17240d46da).

**Pull request merged**